Classes/Core Modules/RPJSRequest.m

```diff
@@ -12,34 +12,114 @@
+        }
     };
-    context[@"__request_post"] = ^(NSString *url, NSDictionary *parameters, JSValue *successCallback, JSValue *errorCallback) {
-        [[RPJSRequest sharedManager] POST:url parameters:parameters success:^(AFHTTPRequestOperation *operation, id responseObject) {
-            NSString *responseString = [[NSString alloc] initWithData:responseObject encoding:NSUTF8StringEncoding];
-            [successCallback callWithArguments:@[ responseString ]];
-        } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
-            [errorCallback callWithArguments:@[ [[error userInfo] description] ]];
-        }];
+    context[@"__request_post"] = ^(JSValue *urlValue, NSDictionary *parameters, JSValue *successCallback, JSValue *errorCallback) {
+        NSString *urlString = [self urlStringFromURLValue:urlValue];
+        NSDictionary *headers = [self headersFromURLValue:urlValue];
+
+        AFHTTPRequestOperation *operation = [self operationWithURLString:urlString httpMethod:@"POST" headers:headers parameters:parameters successCallback:successCallback errorCallback:errorCallback];
+        if (operation) {
+            [[self sharedQueue] addOperation:operation];
+        }
     };
 }
 
 #pragma mark - Private
 
-+ (AFHTTPRequestOperationManager *)sharedManager {
++ (NSOperationQueue *)sharedQueue {
     static dispatch_once_t once;
-    static AFHTTPRequestOperationManager *requestManager;
+    static NSOperationQueue *queue;
     dispatch_once(&once, ^{
-        requestManager = [AFHTTPRequestOperationManager manager];
-        requestManager.responseSerializer = [[AFHTTPResponseSerializer alloc] init];
+        queue = [[NSOperationQueue alloc] init];
+        queue.maxConcurrentOperationCount = NSOperationQueueDefaultMaxConcurrentOperationCount;
+        queue.qualityOfService = NSQualityOfServiceUserInitiated;
     });
-    return requestManager;
+    return queue;
+}
+
++ (AFHTTPRequestOperation *)operationWithURLString:(NSString *)urlString httpMethod:(NSString *)httpMethod headers:(NSDictionary *)headers parameters:(NSDictionary *)parameters successCallback:(JSValue *)successCallback errorCallback:(JSValue *)errorCallback {
+    if (!urlString || urlString.length == 0 || !httpMethod || httpMethod.length == 0) {
+        return nil;
+    }
+
+    AFJSONRequestSerializer *requestSerializer = [[AFJSONRequestSerializer alloc] init];
+    NSError *requestConstructionError;
+    NSMutableURLRequest *request = [requestSerializer requestWithMethod:httpMethod URLString:urlString parameters:parameters error:&requestConstructionError];
+    if (requestConstructionError) {
+        return nil;
+    }
+
+    NSMutableDictionary *mutableHeaders = [headers mutableCopy] ?: [NSMutableDictionary dictionary];
+    if (!mutableHeaders || mutableHeaders.count == 0) {
+        mutableHeaders = [[self defaultHeaders] mutableCopy];
+    }
+    else {
+        // "Extend" user headers dictionary with defaults
```

Comment:
I think this can be simplified using `NSMutableDictionary`s `addEntriesFromDictionary:` method:

```
NSMutableDictionary *headers = [[self defaultHeaders] mutableCopy];
[headers addEntriesFromDictionary:userHeaders];
```

This should allow the user to over write the default headers.